contracts/HoprStake2.sol

```diff
@@ -0,0 +1,443 @@
+    uint256 public totalLocked;  // Total amount of tokens being locked in the incentive program.
+    uint256 public availableReward; // Total amount of reward tokens currently available in the lock.
+
+    // setup ERC1820
+    IERC1820Registry private constant ERC1820_REGISTRY = IERC1820Registry(0x1820a4B7618BdE71Dce8cdc73aAB6C95905faD24);
+    bytes32 private constant TOKENS_RECIPIENT_INTERFACE_HASH = keccak256("ERC777TokensRecipient");
+
+    event Sync(address indexed account, uint256 indexed increment);
+    event Staked(address indexed account, uint256 indexed actualAmount);
+    event Released(address indexed account, uint256 indexed actualAmount);
+    event RewardFueled(uint256 indexed amount);
+    event Redeemed(address indexed account, uint256 indexed boostTokenId, bool indexed factorRegistered);
+    event Claimed(address indexed account, uint256 indexed rewardAmount);
+
+    /**
+     * @dev Provide NFT contract address. Transfer owner role to the new owner address. 
+     * At deployment, it also registers the lock contract as an ERC777 recipient.
+     * @param _nftAddress address Address of the NFT contract.
+     * @param _newOwner address Address of the new owner. This new owner can reclaim any ERC20 and ERC721 token being accidentally sent to the lock contract. 
+     * @param _lockToken address Address of the stake token xHOPR.
+     * @param _rewardToken address Address of the reward token wxHOPR.
+     */
+    constructor(address _nftAddress, address _newOwner, address _lockToken, address _rewardToken) {
+        // implement in favor of testing
+        uint chainId;
+        assembly {
+            chainId := chainid()
+        }
+        if (chainId != 100) {
+            LOCK_TOKEN = _lockToken;
+            REWARD_TOKEN = _rewardToken; 
+            NFT_CONTRACT = IHoprBoost(_nftAddress);
+        }
+        transferOwnership(_newOwner);
+        ERC1820_REGISTRY.setInterfaceImplementer(address(this), TOKENS_RECIPIENT_INTERFACE_HASH, address(this));
+    }
+
+    /**
+     * @dev Given NFT name type and rank name (as in HoprBoost) returns if the hodler has redeemed such an NFT 
+     * @param nftType string Type name of the HoprBoost NFT.
+     * @param nftRank string Rank name of the HoprBoost NFT.
+     * @param hodler address Address of an account that stakes xHOPR tokens and/or redeems its HoprBoost NFT.
+     */
+    function isNftTypeAndRankRedeemed1(string memory nftType, string memory nftRank, address hodler) external view returns (bool) {
+        string memory nftURI = string(abi.encodePacked(nftType, "/", nftRank));
+
+        // compare `boostType/boosRank` of redeemed NFTs with `nftURI`
+        for (uint256 index = 0; index < redeemedNftIndex[hodler]; index++) {
+            uint256 redeemedTokenId = redeemedNft[hodler][index];
+            string memory redeemedTokenURI = NFT_CONTRACT.tokenURI(redeemedTokenId);
+            if (_hasSubstring(redeemedTokenURI, nftURI)) {
```

Comment:
You're not actually checking the NFT Typer here - is that intentional?

Reply:
L.90 `string memory nftURI = string(abi.encodePacked(nftType, "/", nftRank));` constructs a `<nft type>/<nft rank>` string from given input parameters. This string is compared in the loop against `tokenURI` staked NFTs. So Types are checked

Reply:
oh lol, all good then, my bad!